Rectangle.cpp

```diff
@@ -1,15 +1,48 @@
 #ifndef Rectangle_cpp
 #define Rectangle_cpp
+
 #include <iostream>
-#include "Draw.cpp"
+#include "Shape.cpp"
+#include "Area.cpp"
+#include "Input.cpp"
+
 using namespace std;
 
-class Rectangle : public Draw
+class Rectangle : public Shape, public Input
 {
-public:
-    void drawShape() override
-    {
-        cout << "rectangle" << endl;
-    }
+    private:
+        double length = input();
+        double width = input();
+
+    public:
+        void drawShape() override
+        {
+            cout << "rectangle" << endl;
+        }
+
+        double input() override
+        {
+            if (length>0)
+            {
+                cout << "Enter Length" << endl;
+                cin >> length;
+                return length;
+            }
+            else
+            {
+                cout << "Enter Width" << endl;
+                cin >> width;
+                return width;
+            }
+            return 0;
```

Comment:
Seems like redundant case of return. Should either be removed or return a valid value unlike a hard-coded zero.